apollo-router/src/warp_http_server_factory.rs

```diff
@@ -55,10 +55,15 @@ impl HttpServerFactory for WarpHttpServerFactory {
                 .map(|cors_configuration| cors_configuration.into_warp_middleware())
                 .unwrap_or_else(|| Cors::builder().build().into_warp_middleware());
 
-            let routes =
-                get_graphql_request_or_redirect(Arc::clone(&router), Arc::clone(&configuration))
-                    .or(post_graphql_request(router, configuration))
-                    .with(cors);
+            let dispatcher = configuration
+                .subscriber
+                .clone()
+                .map(tracing::Dispatch::new)
+                .unwrap_or_default();
+
+            let routes = get_graphql_request_or_redirect(Arc::clone(&router))
+                .or(post_graphql_request(router))
+                .with(cors);
```

Comment:
Maybe as a followup someday: It seems like a lot of the setup is being done in the async block, while it probably doesn't have to

Reply:
yes, we could move a bit outside of the block